modules/services/gpg-agent.nix

```
@@ -203,10 +203,11 @@ in
         [ cfg.extraConfig ]
       );
 
-      home.sessionVariables =
-        optionalAttrs cfg.enableSshSupport {
-          SSH_AUTH_SOCK = "$(${gpgPkg}/bin/gpgconf --list-dirs agent-ssh-socket)";
-        };
+      home.sessionVariablesExtra = optionalString cfg.enableSshSupport ''
+        if [[ -z "$SSH_AUTH_SOCK" ]]; then
+          export SSH_AUTH_SOCK=$(${gpgPkg}/bin/gpgconf --list-dirs agent-ssh-socket)
```

```suggestion
          export SSH_AUTH_SOCK="$(${gpgPkg}/bin/gpgconf --list-dirs agent-ssh-socket)"
```

In case the socket path has a space in it?

---

Not needed: `export` is special in that regard.

```bash
~ $ export var=$(echo foo bar)
~ $ declare -p var
declare -x var="foo bar"
```

(Doesn't hurt either.)